### DISCUSSION TITLE
How to handle exporting 1 layer while keeping reference to all layers

### DISCUSSION THREAD

**Discussion by dennis-NS** [amzn/style-dictionary#1011] [Q&A]
Given the following example:

```
{
   "primitives": {...},
   "theme-a-semantic": {...},
   "theme-b-semantic": {...},
   "theme-c-semantic": {...},
   "theme-a-components": {...},
   "theme-b-components": {...},
   "theme-c-components": {...}
}
```

I only want to output the values from the component objects. Those component objects can still refer to a semantic object (for it's current theme) or to the primitives layer. Right now i need to supply all the layers otherwise they don't know where to reference to. However this means that style-dictionary makes (in my case css variables) for every token in the file. But i just want my export to have tokens from the components. Am i missing something obvious?

**Comment by pascalduez** [comment]
Hi,

if I understand correctly, you  want to be able to use refrences in your tokens declarations, but not have those references output in the final CSS (custom properties). 

```js
// primitives.json
{
  color: {
    white: {
      value: '#fff',
    }
  }
}

// components.json
{
  button: {
    background: {
      value: '{color.white}',
    }
  }
}
```
Expected ouput:
```css
--button-background: #fff; 
```
This should be controlled by the[ `outputReferences` file options](https://amzn.github.io/style-dictionary/#/formats?id=references-in-output-files) But it's `false` by default.

So I guess the ouput you get is:
```css
--color-white: #fff; 
--button-background: #fff; 
```

Then you will need [filtering](https://amzn.github.io/style-dictionary/#/formats?id=filtering-tokens).

If want to be able to "split" output files [this example](https://github.com/amzn/style-dictionary/tree/main/examples/advanced/matching-build-files) might help. 

Sadly [we don't have access to the file object in filter functions](https://github.com/amzn/style-dictionary/issues/923) at the moment, but you might not need that.

**Reply by dennis-NS** [reply]
Hi!
Yes correct! Only difference is that i have 1 json that has all the components, semantics and the primitives. I'll look into the `outputReferences` that you mentioned, it's not something that i have come accross before and noticed i could use that in this case.
